Documentation/InferOnLinuxKernel.md

```diff
@@ -0,0 +1,159 @@
+- libelf-dev ```sudo apt-get install libelf-dev```  
+- libssl-dev ```sudo apt-get install libssl-dev```
+
+## Prepare to Run Infer on Linux Kernel ## :
+
+Now you can :
+- Compile Linux 4.15 without getting any errors
+- Call infer via command line
+
+To use infer with linux-kernel, first you must capture Linux 4.15 source code with ```infer capture -- make``` command , and then analyse infer-out folder with ```infer analyse```. However with current setup Infer can't capture Linux 4.15 source code. To be able to get an output you need to make some modifications. I explained required modifications below, with two solutions for each of them.
+
+## Modify Infer to mark problematic directories blacklisted ##
+First of all, we need to mark some linux directories as blacklisted, to avoid compilation errors during ```infer capture``` phase.
+To do this you have two options and I prefer second option.
+
+### Modify Infer source code ###
+You can directly modify your infer fork, and mark directories as blacklisted with help of [patch](../scripts/files/0001-Blacklist-files-from-infer-source-code.patch), please don't forget to modify file paths which passed as parameter before applying it.
+
+### Add an inferconfig file ###
+
+Thanks to [@Evan Zhao](https://github.com/Tacinight) we have much simplier and flexible way to solve this problem. You can create an ```.inferconfig``` file in linux-kernel root directory and then mark problematics directories as blacklisted as following: (Please do not forget replace file paths with absolute paths.)
+You can do it via execute ```echo $PWD``` in linux-stable root directory. Then delete first ``` \ ``` and replace ```[linux-stable-absolute-path]```'s with rest of string.
+Example :  
+```"home/ozan/linux-stable/stable/linux-stable/mm"```  
+
+```
+{
+"skip-analysis-in-path": 
+[
+"[linux-stable-absolute-path]/arch/x86/entry/vdso",
+"[linux-stable-absolute-path]/arch/x86/kernel", 
+"[linux-stable-absolute-path]/arch/x86/mm", 
+"[linux-stable-absolute-path]/arch/x86/boot", 
+"[linux-stable-absolute-path]/drivers/acpi", 
+"[linux-stable-absolute-path]/fs", 
+"[linux-stable-absolute-path]/kernel/bpf", 
+"[linux-stable-absolute-path]/mm", 
+"[linux-stable-absolute-path]/net/mac80211"
+]
+}
+```
+You can find why you should these files into the blacklist; and if you dont them, add what kind of error they create at : [Blacklisted-Files-Explanations](../infer/Documentation/BlacklistExplanation) folder.
+
+
+## Set default CC to clang for linux-kernel compilation ##
+
+After that, we need to set clang as a default compiler when we start infer capture phase with ```infer capture -- make``` command.  Otherwise Linux Makefiles uses gcc flags, and since infer compiles with clang, we need clang compiler flags for a successfull compilation.
+For this issue again we have two different solutions, and I prefer second one.
+
+### GCC Hack ###
```

Comment:
We can  put change the CC and HOSTCC as make command line argument while compiling kernel. 
Eg: make CC=clang HOSTCC=clang
Also adding V=1 can make the verbose log.

Reply:
I believe we tried that but `infer capture -- make` modifies the PATH environment and make CC=clang HOSTCC=clang actually did not work solve the problem that infer was called with the wrong compiler flags. We should reconsider this though now that we have infer running properly and possibly CC=clang HOSTCC=clang does in fact work.